File: backend/app/schemas/prompt_template.py
"""管理员提示词 API 的 Pydantic 模型。"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class PromptTemplateAdminOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    layer: str
    template_text: str
    applicable_mode: str
    is_active: bool
    version: int
    created_at: datetime | None = None
    updated_at: datetime | None = None


class PromptTemplateLayerGroup(BaseModel):
    """同一 layer 下按 applicable_mode 分桶。"""

    layer: str
    by_mode: dict[str, list[PromptTemplateAdminOut]] = Field(
        default_factory=dict,
        description="键为 strict / creative / all 等",
    )


class PromptTemplatesGroupedResponse(BaseModel):
    layers: list[PromptTemplateLayerGroup]


class PromptTemplateCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    layer: str = Field(..., min_length=1, max_length=20)
    template_text: str = Field(..., min_length=1)
    applicable_mode: str = Field(default="all", max_length=10)
    is_active: bool = True


class PromptTemplateUpdate(BaseModel):
    name: str | None = Field(None, max_length=50)
    template_text: str | None = None
    applicable_mode: str | None = Field(None, max_length=10)
    is_active: bool | None = None
    bump_version: bool = False


class PromptTemplatePutResponse(PromptTemplateAdminOut):
    pass

File: frontend/src/types/promptTemplate.ts
/** 与后端 app/schemas/prompt_template.py 对齐 */

export interface PromptTemplateAdminOut {
  id: number;
  name: string;
  layer: string;
  template_text: string;
  applicable_mode: string;
  is_active: boolean;
  version: number;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface PromptTemplateLayerGroup {
  layer: string;
  by_mode: Record<string, PromptTemplateAdminOut[]>;
}

export interface PromptTemplatesGroupedResponse {
  layers: PromptTemplateLayerGroup[];
}

export interface PromptTemplateUpdate {
  name?: string | null;
  template_text?: string | null;
  applicable_mode?: string | null;
  is_active?: boolean | null;
  bump_version?: boolean;
}

export interface PromptTemplateCreate {
  name: string;
  layer: string;
  template_text: string;
  applicable_mode?: string;
  is_active?: boolean;
}
